src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceAuthorizationBehavior.cs

```diff
@@ -154,8 +176,17 @@ private void CopyAuthorizationPoliciesAndManager(ServiceAuthorizationBehavior ot
             _serviceAuthorizationManager = other._serviceAuthorizationManager;
         }
 
-        void IServiceBehavior.Validate(ServiceDescription description, ServiceHostBase serviceHostBase)
+        void IServiceBehavior.Validate(ServiceDescription serviceDescription, ServiceHostBase serviceHostBase)
         {
+            foreach (ServiceEndpoint endpoint in serviceDescription.Endpoints)
+            {
+                TransportBindingElement transportBindingElement = endpoint.Binding.CreateBindingElements().Find<TransportBindingElement>();
+                if (transportBindingElement != null)
+                {
```

Comment:
A binding is required to have a TransportBindingElement. I just looked up the code to link it and discovered we have a bug. ServiceEndpoint.EnsureInvariants is supposed to ball Binding.EnsureInvariants but that's been commented out. Subject to that bug being fixed (should just require uncommenting the one line) you should be able to presume transportBindingElement is not null. I can see an Fx.Assert being useful here still, to catch bugs.

Reply:
done here https://github.com/CoreWCF/CoreWCF/pull/850/commits/87b5c264ab6c606dcabe988487ff9966b939f639